apps/api/src/app/workflows-v2/generate-preview.e2e.ts

```diff
@@ -189,55 +309,155 @@ describe('Generate Preview', () => {
       });
     });
 
-    describe('Error Handling', () => {
-      describe('In App Partial control values', () => {
-        it('Should not fail if inApp is providing partial URL in redirect', async () => {
-          const steps = [{ name: 'IN_APP_STEP_SHOULD_NOT_FAIL', type: StepTypeEnum.IN_APP }];
-          const createDto = buildCreateWorkflowDto('', { steps });
-          const novuRestResult = await workflowsClient.createWorkflow(createDto);
-          if (!novuRestResult.isSuccessResult()) {
-            throw new Error('should create workflow');
-          }
-          const controlValues = {
-            subject: `{{subscriber.firstName}} Hello, World! ${PLACEHOLDER_SUBJECT_INAPP}`,
-            body: `Hello, World! {{payload.placeholder.body}}`,
-            avatar: 'https://www.example.com/avatar.png',
-            primaryAction: {
-              label: '{{payload.secondaryUrl}}',
-              redirect: {
-                target: RedirectTargetEnum.BLANK,
-              },
-            },
-            secondaryAction: null,
+    describe('payload sanitation', () => {
+      it('Should produce a correct payload when pipe is used etc {{payload.variable | upper}}', async () => {
+        const { stepDatabaseId, workflowId } = await createWorkflowAndReturnId(workflowsClient, StepTypeEnum.SMS);
+        const requestDto = {
+          controlValues: {
+            body: 'This is a legal placeholder with a pipe [{{payload.variableName | upcase}}the pipe should show in the preview]',
+          },
+        };
+        const previewResponseDto = await generatePreview(
+          workflowsClient,
+          workflowId,
+          stepDatabaseId,
+          requestDto,
+          'email'
+        );
+        expect(previewResponseDto.result!.preview).to.exist;
+        if (previewResponseDto.result!.type !== 'sms') {
+          throw new Error('Expected sms');
+        }
+        expect(previewResponseDto.result!.preview.body).to.contain('{{PAYLOAD.VARIABLENAME | UPCASE}}');
+        expect(previewResponseDto.previewPayloadExample).to.exist;
+        expect(previewResponseDto?.previewPayloadExample?.payload?.variableName).to.equal(
+          '{{payload.variableName | upcase}}'
+        );
```

Comment:
This is not a suggestion; in this comment, I want to highlight current potential issues we are facing. 

Currently, the code is extracting and building the payload schema (used for validation and sanitation in `PrepareAndValidateContentUsecase`) based on the new variable extraction powered by LiquidJS. This is important because we should rely more on LiquidJS, and this is a good example of why.

In the code below, my suggestion is an old test we had that used an invalid filter not supported by LiquidJS, leading to false positive test results.

```suggestion
            body: 'This is a legal placeholder with a pipe [{{payload.variableName | upper}}the pipe should show in the preview]',
          },
        };
        const previewResponseDto = await generatePreview(
          workflowsClient,
          workflowId,
          stepDatabaseId,
          requestDto,
          'email'
        );
        expect(previewResponseDto.result!.preview).to.exist;
        if (previewResponseDto.result!.type !== 'sms') {
          throw new Error('Expected sms');
        }
        expect(previewResponseDto.result!.preview.body).to.contain('{{payload.variableName | upper}}');
        expect(previewResponseDto.previewPayloadExample).to.exist;
        expect(previewResponseDto?.previewPayloadExample?.payload?.variableName).to.equal(
          '{{payload.variableName | upper}}'
        );
```